tests/main_tests/test_data_check.py

```diff
@@ -0,0 +1,49 @@
+"""
+This script tests the data check function. This is a test.
+"""
+import pathlib
+
+import numpy as np
+import pandas as pd
+from pandas.testing import assert_frame_equal
+
+from src.cytocherrypick.generate_test_data import data_generation, data_writing
+
+
+def test_data_check():
+    """
+    check to ensure that the test data is generated correctly
+    """
+    num_of_rows = 100
+    file_out_path = "tests/data/test_data.csv"
+    df = data_generation(output_file_path=file_out_path, num_of_rows=num_of_rows)
```

Comment:
Consider making data generation a [Pytest fixture](https://docs.pytest.org/en/7.4.x/how-to/fixtures.html#how-to-fixtures) within a `conftest.py` file which could be used by one or many other tests.